library/network_connections.py

```diff
@@ -334,6 +334,8 @@ def ifcfg_create(
             ifcfg["ONBOOT"] = "no"
 
         ifcfg["DEVICE"] = connection["interface_name"]
+        if connection["cloned_mac"] and connection["cloned_mac"] != "default":
```

Comment:
With initscripts provider, `connection["cloned_mac"]` will not be 'default' at this step, I think that we can remove it and only keep it as `if connection["cloned_mac"]:`.